### ISSUE TITLE
Function from dates to general linelist censored primary and secondary times

### ISSUE THREAD

**Issue by athowes** [epinowcast/epidist#142]
After completion of intial PR https://github.com/epinowcast/epidist/pull/125 adding `as_latent_individual` functionality we should continue to build out the range of data which can be processed and run with the `latent_individual` model. To enable date format data to be processed I propose to add a helper function mapping it to a format which can be accepted by existing `as_latent_individual` functionality. More specifically:

A. General format (as accepted by current `as_latent_individual`):

| column | type |
|---------|-----|
| case | int |
| ptime_lwr | num |
| ptime_upr | num |
| stime_lwr | num |
| stime_upr | num |
| obs_at | num |

B. Only dates format:

| column | type |
|---------|-----|
| case | int |
| ptime_date | num or date |
| stime_date | num or date |
| obs_at | num |

Proposed workflow would be:

```
A <- function_in_this_issue(B)
prep_obs <- as_latent_individual(A)
```

## Maybe or out of scope for this issue

* A date is just one example of a fixed censoring window. Can we have other fixed censoring windows work with this function? I think should be possible but would call this "nice-to-have"
* Things to do with models outside `latent_individual`. Maybe this is a mistake and need to think more about how useful moving to type A will be in general?

## Uncertainties

Could argue about this approach versus `prep_obs <- function_in_this_issue(B)` direct routes? i.e. putting the transformation inside `as_latent_individual`.

## New issues

* List more possible formats i.e. C and D
* Scope out other C -> A or D -> A functions

**Comment by athowes** [comment]
Sam notes that there are multiple possible date formats. For example there could be start and end points which are dates.

**Comment by athowes** [comment]
Note that this will be closed by #390.

**Comment by seabbs** [comment]
Done done woo
